Runtime/Core/InstalledJavaVersionCheck/InstalledJavaVersionProvider.cs

```diff
@@ -29,79 +36,44 @@ public CheckInstalledJavaVersionResponse CheckInstalledJavaVersion(CheckInstalle
-
-            Match outputMatch = outputPattern.Match(processOutput);
-
-            if (!outputMatch.Success)
-            {
-                return Response.Ok(new CheckInstalledJavaVersionResponse
-                {
-                    IsInstalled = false
-                });
+                return makeResponse(false);
             }
 
-            // example output: java version "1.8.0"
-            var versionLine = outputMatch.Groups[0].ToString();
-            string[] outputWords = versionLine
-                .Split(' ');
-
-            if (versionLine.Length < 3)
-            {
-                return Response.Ok(new CheckInstalledJavaVersionResponse
-                {
-                    IsInstalled = false
-                });
+            if (processOutput == null) {
+                return makeResponse(false);
             }
 
-            string installedVersion = outputWords[2]
-                .Replace("\"", "");
+            // Expected output format:
+            //  (java|openjdk) version "<majorVersion>.<minorVersion>.<build>"
+            //  ex: java version "1.8.0_291"
+            //      openjdk version "1.8.0_322"
+            //      openjdk version "19"
+            //  majorVersion and minorVersion are integers. build can be anything
+            var outputPattern = new Regex("(?:openjdk|java) version \"(?<majorVersion>\\d+)(?:\\.(?<minorVersion>\\d+)(?:\\.[^\"]+))?\"");
 
-            var regex = new Regex(@"([0-9]+)");
-            MatchCollection versionNumbers = regex.Matches(installedVersion);
+            Match outputMatch = outputPattern.Match(processOutput);
 
-            if (versionNumbers.Count < 2)
-            {
-                return Response.Ok(new CheckInstalledJavaVersionResponse
-                {
-                    IsInstalled = false
-                });
+            if (!outputMatch.Success) {
+                return makeResponse(false);
             }
 
-            //we need to match major version in string like "1.8.0" here major version is 8
-            string majorVersion = versionNumbers[1].Value;
-
-            bool majorVersionParsed = int.TryParse(majorVersion, out int majorVersionAsNumber);
-
-            if (!majorVersionParsed)
-            {
-                return Response.Ok(new CheckInstalledJavaVersionResponse
-                {
-                    IsInstalled = false
-                });
-            }
+            //  if majorVersion is 1, we use minorVersion as the majorVersion, since java version had the format 1.?? until java 8
+            var majorVersion = outputMatch.Groups["majorVersion"].ToString();
+            var minorVersion = outputMatch.Groups["minorVersion"].ToString();
+            var actualMajorVersion = majorVersion.Equals("1") ? minorVersion : majorVersion;
 
+            bool majorVersionParsed = int.TryParse(actualMajorVersion, out int majorVersionAsNumber);
```

Comment:
`bool majorVersionParsed` is not used and could be removed (unless parsing could fail, but since we are explicitly matchking for `\d`, it shouldn't)